main.c

```diff
@@ -160,6 +175,9 @@ main(int argc, char *argv[])
 		GETOPT_OPT("-f"):
 			force_resources = 1;
 			break;
+		GETOPT_OPTARG("--logN"):
+			GETOPT_PARSENUM_WITHIN(&params.logN, 10, 32);
```

Comment:
I'd be inclined to make this range 10--40, just for future-proofing.

We're applying the "make sure we have a reasonable amount of memory/time" tests (assuming no `-f`) right?  So there should already be a seatbelt to protect against crazy parameters.

Reply:
No, there's no "reasonable" checks.  `scryptenc_setup:`

```
        /* Determine parameters. */
        if (P->logN != 0) {
                /* Check logN, r, p, without any resource limits. */
                if ((rc = checkparams(0, 0, 0, P->logN, P->r, P->p,
                    verbose, 1)) != 0) {
```

The idea is that if somebody explicitly specifies N, r, p, then we assume that they know what they're doing.

(this change was in a previous PR, in case you're wondering why you didn't see that code in this PR)

Reply:
* the "without any resource limits" refers to the final `1` in `checkparams()`, which enables `force`.